app/controllers/lister.rb

```diff
@@ -0,0 +1,57 @@
+enable :sessions
+
+get "/" do
+  if session[:logged_in] != true
+    erb :home
+  else
+    redirect '/dashboard?search= '
+  end
+end
+
+get '/dashboard' do
+    @search = search_recipes(params[:search])
+  erb :dashboard
+end
+
+get '/create/user' do
+  erb :create_user
+end
+
+post  '/add/:recipe_id/:search' do
+  User.find(session[:user_id]).recipes.create(food2fork_id: params[:recipe_id])
+  redirect '/dashboard?search=' + params[:search]
+end
+
+get '/my/list' do
+  @list = User.find(session[:user_id]).lists
+
+  user_recipes = User.find(session[:user_id]).recipes
+  @recipe_data = []
+
+  user_recipes.each do |recipe|
+    @recipe_data << return_recipe_data(recipe.food2fork_id)
+  end
+
+  erb :my_list
+end
+
+delete '/my/list/delete/:recipe_id' do
+  User.find(session[:user_id]).recipes.find_by(food2fork_id: params[:recipe_id]).destroy
+  redirect '/my/list'
+end
+
+
+put '/my/list/add/:recipe_id' do
+  return_recipe_data(params[:recipe_id])['ingredients'].each do |list_item|
+    User.find(session[:user_id]).lists.create(ingredients: list_item)
```

Comment:
You also need to make sure the create worked. Either user `create!` or check the return value.